strapi/strapi
# Fix duplicate sort results

### What does it do?

In query builder when we query for data and we have joins, we run `qb.distinct` which should remove duplicate data from the query but after some testing it doesn't seem to affect queries with multiple joins and duplicate data was queried. So I opted to use `qb.groupBy` instead when we have multiple joins while passing it the referencedColumn (id) of the firstJoin.

### Why is it needed?

When doing nested sorting, duplicate data is being returned

### How to test it?

run a strapi application and create content types with a relation, try to sort the parent content type based on a relation field there should be no duplicates, you can follow the same method in the issue I will reference below.

### Related issue(s)/PR(s)

related issue #11892
